Nodejs/Product/Analysis/Analysis/JsAnalyzer.cs

```diff
@@ -140,8 +140,13 @@ internal partial class JsAnalyzer : IGroupableAnalysisProject, IDeserializeIniti
             }
 
             var tree = Modules.GetModuleTree(Path.GetDirectoryName(filePath));
-            
+
             tree.DefaultPackage = entryPoint;
+            if (dependencies != null) {
+                foreach (var dependency in dependencies) {
+                    RequireAnalysisUnit.UpdateVisibilities(tree, Modules, new ProjectEntry(this, filePath, null), dependency);
```

Comment:
the "new ProjectEntry(" should at least be moved out of the loop...  I wonder if you could get rid of it entirely by using a different ProjectEntry but it may not be worth it.